src/client/inputhandler.cpp

```diff
@@ -112,56 +114,77 @@ bool MyEventReceiver::OnEvent(const SEvent &event)
 		const KeyPress &keyCode = event.KeyInput;
 		if (keysListenedFor[keyCode]) {
 			if (event.KeyInput.PressedDown) {
+				// If the key is being held down then the OS may
+				// send a continuous stream of keydown events.
+				// In this case, we don't want to let this
+				// stream reach the application as it will cause
+				// certain actions to repeat constantly.
+				if (!IsKeyDown(keyCode)) {
```

Comment:
I wonder if this check is redundant. Aren't events only sent if there are state changes? Pretty sure a key press event is sent only if the key was just pressed, and isn't continuously sent as long as the key is pressed.

Reply:
Apparently not - as long as the key is pressed, one event is sent, and then a stream of events are sent until the key is released; just as this helpful comment suggests. But... why? :thinking:

Reply:
Is this related to an operating system's key repeat settings?  Just curious.

Reply:
It seems like it; *shouldn't* be, though...